src/utils/sharedbuffergeometry.js

```diff
@@ -10,29 +9,21 @@ export default class SharedBufferGeometry {
     this.addBuffer();
   }
 
-  restartPrimitive() {
-    if (this.idx.position >= this.current.attributes.position.count) {
-      console.error("maxBufferSize limit exceeded");
-    } else if (this.idx.position !== 0) {
-      let prev = (this.idx.position - 1) * 3;
-      const position = this.current.attributes.position.array;
-      this.addVertex(position[prev++], position[prev++], position[prev++]);
-
-      this.idx.color++;
-      this.idx.normal++;
-    }
-  }
-
   remove(prevIdx, idx) {
     // Loop through all the attributes: position, color, normal,...
     if (this.idx.position > idx.position) {
       for (const key in this.idx) {
-        const componentSize = 3;
+        const componentSize = key === "index" ? 1 : 3;
         let pos = prevIdx[key] * componentSize;
         const start = (idx[key] + 1) * componentSize;
         const end = this.idx[key] * componentSize;
         for (let i = start; i < end; i++) {
-          this.current.attributes[key].array[pos++] = this.current.attributes[key].array[i];
+          if (key === "index") {
+            //index needs to be handled specially
+            this.current.index.array[pos++] = this.current.index.array[i] - idx.position - 1;
```

Comment:
What does this line do / why are indices handled differently? I'm having a hard time keeping track of `idx`, `prevIdx`, `this.idx` and all of their `.index` attributes

Reply:
unlike `position`, `normals` or `colors`, indexes are not accessible on the `geometry.attributes` property, they're instead accessed via `geometry.index` directly, or via `getIndex()` or `setIndex()`